src/coreComponents/finiteElement/elementFormulations/Qk_Hexahedron_Lagrange_GaussLobatto.hpp

```diff
@@ -923,6 +968,57 @@ calcGradN( localIndex const q,
   return calcGradN( q, X, gradN );
 }
 
+template< typename GL_BASIS >
+GEOS_HOST_DEVICE
+GEOS_FORCE_INLINE
+real64
+Qk_Hexahedron_Lagrange_GaussLobatto< GL_BASIS >::calcGradNWithCorners( localIndex const q,
+                                                                       real64 const (&X)[8][3],
+                                                                       real64 (& gradN)[numNodes][3] )
+{
+  int qa, qb, qc;
+  GL_BASIS::TensorProduct3D::multiIndex( q, qa, qb, qc );
+
+  real64 J[3][3] = {{0}};
+
+  jacobianTransformation( qa, qb, qc, X, J );
+
+  real64 const detJ = LvArray::tensorOps::invert< 3 >( J );
+
+  applyTransformationToParentGradients( q, J, gradN );
+
+  return detJ;
+}
+//*************************************************************************************************
+template< typename GL_BASIS >
+GEOS_HOST_DEVICE
+GEOS_FORCE_INLINE
+real64
+Qk_Hexahedron_Lagrange_GaussLobatto< GL_BASIS >::calcGradNWithCorners( real64 const (&coords)[3],
+                                                                       real64 const (&X)[8][3],
+                                                                       real64 (& gradN)[numNodes][3] )
+{
+  real64 J[3][3] = {{0}};
+
+  jacobianTransformation( coords, X, J );
```

Comment:
I think you should use the other version of jacobianTransformation here, the one with qa, qb and qc.
Because if not the compilation crash because the compiler tries to put the value of numNodes to every possible values depending the orders (see the crahs inside the CIà

Reply:
You're right, I'll add a version for the calculation of the jacobian at an arbitrary point that does not require the location of high-order nodes